Issue/10638 material sharing settings

khaykov opened the pull request:
Fixes #10638 

This PR moves Publicize screen to Material Theme and adds Dark Theme support. There are some code changes for cosmetic reasons.

While refactoring is not in the scope of this PR, since I touched those files, I cleaned up/optimized some of the redundant code as suggested by AS, so do not be alarmed by the size of the PR. I suggest turning on "Hide whitespace changes" option during the review.

There is a huge amount of PRs in this feature that covers all the screens within the app, so I'm requesting reviews from a wide variety of people.

![comparison_sharing](https://user-images.githubusercontent.com/728822/70952265-8c1a6500-201a-11ea-88c3-d06c4dbcb9d9.png)

To test:
- Navigate to Publicize screen and confirm it's visual and functional integrity in both Light and Dark themes. The theme can be changed from App Settings.

PR submission checklist:

- [x] I have considered adding unit tests where possible.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

jkmassel commented:
I'm moving this PR to 14.1 since we're code freezing today.

khaykov commented:
@mattmiklic Sure! I moved everything out of cards. Had to adjust margins a bit :)

[![Image from Gyazo](https://i.gyazo.com/44613dd4a3841ff72e570f9a2af83673.png)](https://gyazo.com/44613dd4a3841ff72e570f9a2af83673)
[![Image from Gyazo](https://i.gyazo.com/8c9dc7ac2616d67e13e858c8464e928c.png)](https://gyazo.com/8c9dc7ac2616d67e13e858c8464e928c)
[![Image from Gyazo](https://i.gyazo.com/3ed3b7dbba1d664391f882dcedd975bf.png)](https://gyazo.com/3ed3b7dbba1d664391f882dcedd975bf)
[![Image from Gyazo](https://i.gyazo.com/421e920fb4c6dc1c22f6ab30fcda7b28.png)](https://gyazo.com/421e920fb4c6dc1c22f6ab30fcda7b28)

mattmiklic commented:
That's a nice improvement; thanks for doing that!

oguzkocer commented on WordPress/src/main/java/org/wordpress/android/widgets/WPPrefView.java:
```diff
@@ -431,18 +419,14 @@ private Dialog getCheckListDialog() {
-                        doPrefChanged();
-                    }
+                .setPositiveButton(android.R.string.ok, (dialog, which) -> {
+                    SparseBooleanArray checkedItems1 =
```
Nitpick: Should we name `checkedItems1` to something else? Or maybe we could do something like this:

```
mListItems.setCheckedItems(((AlertDialog) dialog).getListView().getCheckedItemPositions());
```

_P.S: Android Studio can't find `MaterialAlertDialogBuilder` for me, so I couldn't play with it._

khaykov replied:
Good catch! AS does it sometimes with optimizing code.

khaykov commented on WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeAccountChooserListAdapter.java:
```diff
@@ -38,6 +39,7 @@ public PublicizeAccountChooserListAdapter(Context context, List<PublicizeConnect
         mSelectedPosition = 0;
     }
 
+    @NotNull
```
Got it! Was not sure about this :)

khaykov commented on WordPress/src/main/java/org/wordpress/android/ui/publicize/ConnectButton.java:
```diff
@@ -15,6 +16,7 @@
  */
 public class ConnectButton extends FrameLayout {
```
Agree, that's a good idea 👍

khaykov commented on WordPress/src/main/java/org/wordpress/android/ui/publicize/ConnectButton.java:
```diff
@@ -68,4 +70,10 @@ public void setAction(ConnectAction newAction) {
+    @Override
+    public void setOnClickListener(@Nullable OnClickListener l) {
+        super.setOnClickListener(l);
+        mConnectButton.setOnClickListener(l);
```
Right, we probably don't need super here. I changed the base class and removed this method all along.

khaykov commented:
@oguzkocer thank you so much for checking Dark Mode PR's! Really appreciate that 🙇 I hope I addressed your concerns.

Regarding layout changes, I agree! I don't think there is a good way to approach this apart from looking at the layout.
